crates/router/src/connector/utils.rs

```diff
@@ -620,6 +621,16 @@ impl PaymentsPreProcessingData for types::PaymentsPreProcessingData {
             .clone()
             .ok_or_else(missing_field_err("complete_authorize_url"))
     }
+    fn connector_mandate_id(&self) -> Option<String> {
+        self.mandate_id
+            .as_ref()
+            .and_then(|mandate_ids| match &mandate_ids.mandate_reference_id {
+                Some(payments::MandateReferenceId::ConnectorMandateId(connector_mandate_ids)) => {
+                    connector_mandate_ids.connector_mandate_id.clone()
+                }
+                _ => None,
```

Comment:
Can we avoid the wildcard match arm, and expand it instead?

Reply:
Will be picking this up in this PR https://github.com/juspay/hyperswitch/pull/4884